api/tests/test_utils/db_utils.py

```diff
@@ -0,0 +1,195 @@
+    # merged_df_1 = pd.merge(dfs['validationreport'], dfs['validationreportgtfsdataset'], left_on='id',
+    #                        right_on='validation_report_id')
+
+    datasets = dfs["gtfsdataset"]
+    dataset_id_to_stable_id = datasets.set_index("id")["stable_id"].to_dict()
+
+    # Replace feed_id with stable_id in the gtfsdataset DataFrame
+    datasets["feed_stable_id"] = datasets["feed_id"].replace(feed_id_to_stable_id)
+    datasets["id"] = datasets["stable_id"]
+    dfs["datasets"] = datasets.drop(columns=["feed_id", "stable_id"])
+
+    validationreports = dfs["validationreport"]
+    validationreports["name"] = ["vr_" + str(i + 1) for i in range(len(validationreports))]
+    validationreport_id_to_name = validationreports.set_index("id")["name"].to_dict()
+    validationreports["id"] = validationreports["id"].replace(validationreport_id_to_name)
+
+    validationreports["dataset_id"] = validationreportgtfsdatasets["dataset_id"].replace(dataset_id_to_stable_id)
+    validationreports["html_report"] = "someurl"
+    validationreports["json_report"] = "someurl"
+    dfs["validation_reports"] = validationreports.drop(columns=["name"])
+    del dfs["validationreport"]
+
+    features = dfs["feature"]
+
+    # Extract the 'name' values into a list
+    name_list = features["name"].values
+
+    # Convert the list into a DataFrame
+    dfs["features"] = pd.DataFrame(name_list)
+
+    del dfs["feature"]
+
+    featurevalidationreports = dfs["featurevalidationreport"]
+
+    featurevalidationreports["validation_report_id"] = featurevalidationreports["validation_id"].replace(
+        validationreport_id_to_name
+    )
+    featurevalidationreports["feature_name"] = featurevalidationreports["feature"]
+    dfs["validation_report_features"] = featurevalidationreports.drop(columns=["validation_id", "feature"])
+    del dfs["featurevalidationreport"]
+
+    notices = dfs["notice"]
+    notices["dataset_id"] = notices["dataset_id"].replace(dataset_id_to_stable_id)
+    notices["validation_report_id"] = notices["validation_report_id"].replace(validationreport_id_to_name)
+    dfs["notices"] = notices
+    del dfs["notice"]
+
+    del dfs["feedreference"]
+    del dfs["location"]
+    del dfs["locationfeed"]
+    del dfs["externalid"]
+    del dfs["gtfsdataset"]
+    del dfs["redirectingid"]
+    del dfs["validationreportgtfsdataset"]
+    del dfs["feed"]
+
+    for table, df in dfs.items():
+        if df.shape[0] > 0:
+            # Convert the DataFrame into a JSON object and add it to the data dictionary
+            records = df.to_dict(orient="records")
+            data[table] = [{k: v for k, v in record.items() if v is not None} for record in records]
+
+    # Write the data dictionary to a JSON file
+    with open(file_name, "w") as f:
+        json.dump(data, f, cls=CustomEncoder)
+
+
+def is_localhost(url):
+    return url is None or url.host == "localhost"
+
+
+def empty_database(db, url):
```

Comment:
Why not use a more "generic" way to avoid updating the table list whenever there is a change in the database, [example](https://github.com/MobilityData/mobility-feed-api/blob/d1e236714778da68893dc78c914e02fba21374a0/functions-python/test_utils/database_utils.py#L60)

Reply:
Sure. I did not know about clean_testing_db().
On problem I encountered that led to a series of hardcoded deletes is that it has to be done in the proper order. It looked simpler just to do it this way.
I will change it to use clean_testing_db()